connectors/sources/google_cloud_storage.py

```diff
@@ -223,12 +269,12 @@ async def fetch_buckets(self):
         Yields:
             Dictionary: Contains the list of fetched buckets from Google Cloud Storage.
         """
-        async for bucket in self._api_call(
+        async for bucket in self.get_storage_client()._api_call(
             resource="buckets",
             method="list",
             full_response=True,
-            project=self.user_project_id,
-            userProject=self.user_project_id,
+            project=self.google_storage_client.user_project_id,
```

Comment:
The `user_project_id` seems to be already set inside the client, right? Do we need to pass it explicitly here or can we use the field inside `_api_call`?

Reply:
Right we can have `user_project_id` in the client class itself, but it is passed as value to many different parameters depending on the API we are trying to hit. To accommodate this in the client class, I would have to have checks on the API and then assign the `user_project_id` to the respective parameter.
To avoid multiple if-else ladder, I considered using it as a parameter to `_api_call()`.
WDYT? Should I keep it the same way?

Reply:
Hey @timgrein , suggested changes are not scope of this PR. This was raised to parse JSON from the Kibana UI.
Feel free to create a new Github issue where we can continue the discussion on the new architecture.
We should not block this PR to merge for the enhancement suggested. WDYT?

Reply:
I see, could you still update `api_call` removing the leading underscore, happy to approve afterwards 👍